include/grpc++/auth_context.h

```diff
@@ -34,15 +34,42 @@
 #ifndef GRPCXX_AUTH_CONTEXT_H
 #define GRPCXX_AUTH_CONTEXT_H
 
+#include <iterator>
 #include <vector>
 
 #include <grpc++/config.h>
 
+struct grpc_auth_context;
```

Comment:
Do you think we could put the iterator in another header? I think that would be cleaner. At least we would not "pollute" the AuthContext header (even though we would include the new iterator header).